### PR TITLE
Blaze: Add banner on My Store tab

### LINKED ISSUE #10157: Add the new banner on the Dashboard screen.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #10157 

⚠️  Please make sure #10135 is reviewed and merged.

## Description

This PR adds the Blaze banner to the My Store tab. Changes include:
- Updated `StoreOnboardingView` to remove the top and bottom spacer in collapsed mode and added spacing to the `DashboardViewController`'s container stack view instead.
- Since the banner should only be displayed when the store has at least one product, we need to check for products in the store when determining the banner visibility. We have an action in `ProductAction` for that check, but it was named `checkProductsOnboardingEligibility`. To make the action more generic, I renamed it to `checkIfStoreHasProducts` and updated the logic of product onboarding banner accordingly.
- Updated the `DashboardViewModel` to check for the visibility of the Blaze banner and support hiding it when necessary.
- Updated the `DashboardViewController` to:
  - insert the Blaze banner below the store onboarding card or the header view
  - recheck the Blaze banner upon pull-to-refresh.
- Also: updated `ProductsViewController` to hide the Blaze banner if the product list is updated between the empty and non-empty states.

## Testing instructions

- Log in as an admin to a public WPCom site with at least one product to ensure you are eligible to create a Blaze campaign.
- If the site has no products, notice that the product onboarding banner is displayed and the Blaze banner is not. The Blaze banner should not be displayed on the product list screen either.
- Add a product to the store and refresh the dashboard screen, the Blaze banner should be displayed on both the product list screen and the dashboard screen.
- If the store still has incomplete onboarding tasks, the Blaze banner should be displayed below the onboarding card.
- Filter "blaze" in Xcode console, notice that the source for the events on the My Store tab is `my_store`.

## Screenshots

| With JITM banner | With onboarding card | Without other banners |
| ----- | ----- | ----- |
|  |  |  |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/ProductStore.swift [hunk 1]
```diff
@@ -495,21 +495,21 @@ private extension ProductStore {
         upsertStoredProductsInBackground(readOnlyProducts: [product], siteID: product.siteID, onCompletion: onCompletion)
     }
 
-    /// Checks if the store is eligible for products onboarding.
+    /// Checks if the store already has any products.
     /// Returns `true` if the store has no products.
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/ProductStore.swift]: I think we have to update this comment.

File: Yosemite/Yosemite/Actions/ProductAction.swift [hunk 2]
```diff
@@ -107,10 +107,10 @@ public enum ProductAction: Action {
     ///
     case replaceProductLocally(product: Product, onCompletion: () -> Void)
 
-    /// Checks if the store is eligible for products onboarding.
+    /// Checks if the store already has any products.
     /// Returns `true` if the store has no products.
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Actions/ProductAction.swift]: Same here. I think we have to update this comment.

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift [hunk 3]
```diff
@@ -609,6 +609,143 @@ final class DashboardViewModelTests: XCTestCase {
...
+            default:
+                break
+            }
+        }
+        await viewModel.updateBlazeBannerVisibility()
+
+        //  Then
+        XCTAssertTrue(viewModel.showBlazeBanner)
+    }
+    // swiftlint:enable:next line_length
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift]: We don't have to enable it again as you are using `swiftlint:disable:next` to disable it only for the next line.

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift [hunk 4]
```diff
@@ -290,6 +284,47 @@ final class DashboardViewModel {
...
+    ///
+    @MainActor
+    func updateBlazeBannerVisibility() async {
+        showBlazeBanner = await isBlazeBannerVisible()
+    }
+
+    private func isBlazeBannerVisible() async -> Bool {
+        async let isSiteEligible = blazeEligibilityChecker.isSiteEligible()
+        async let storeHasProducts = (try? checkIfStoreHasProducts(siteID: siteID)) ?? false
+        guard (await isSiteEligible, await storeHasProducts) == (true, true) else {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift:299]: Nice work sending these requests in parallel.

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift [hunk 5]
```diff
@@ -609,6 +609,143 @@ final class DashboardViewModelTests: XCTestCase {
         // Then
         XCTAssertNil(analyticsProvider.receivedEvents.firstIndex(of: "dashboard_store_timezone_differ_from_device"))
     }
+
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift:612]: Thanks for adding these unit tests. 💯

**Review by selanthiraiyan — APPROVED** [review]
Works great. 🚀 

I just left a couple of non-blocking minor nits.

File: Yosemite/Yosemite/Stores/ProductStore.swift (hunk 1 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/ProductStore.swift]: Nice catch! Updated in 339bae8c8a10318bd06041ed06baed8dc30129ae.

File: Yosemite/Yosemite/Actions/ProductAction.swift (hunk 2 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Actions/ProductAction.swift]: Also updated in 339bae8c8a10318bd06041ed06baed8dc30129ae.

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift (hunk 3 above)
Comment (itsmeichigo) [reply WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift]: TIL! Removed in 8bab03bc8b9240ba7bcd729c9a20fd9881fe0fcc.

**Pull request merged**